Source/WebCore/dom/NodeIterator.cpp

```diff
@@ -159,20 +161,20 @@ void NodeIterator::updateForNodeRemoval(Node& removedNode, NodePointer& referenc
 
     // Iterator is not affected if the removed node is the reference node and is the root.
     // or if removed node is not the reference node, or the ancestor of the reference node.
-    if (!removedNode.isDescendantOf(root()))
+    if (!removedNode.isDescendantOf(protectedRoot()))
```

Comment:
`protectedRoot()` is called a few times in this function, maybe we want to cache in a local variable to avoid ref counting churn.

Reply:
Sure.